source/rafcon/gui/helpers/state_machine.py

```diff
@@ -587,9 +590,13 @@ def save_state_machine_as(path=None, recent_opened_notification=False, as_copy=F
             logger.error("No function defined for creating a folder")
             return False
         folder_name = selected_state_machine_model.state_machine.root_state.name
+        library_os_path, _, _, _ = library_ctrl.extract_library_properties_from_selected_row()
+        if library_os_path is not None and library_os_path.startswith('[source]:\n'):
```

Comment:
What is this about? When does the `library_os_path` start with `[source]`?

Reply:
The "library_os_path" starts with "[source]" for the root library keys path stored in the library tree. In contrast, "library_os_path" starts without "[source]" for the library's path stored in the library tree.

It was like that from the beginning, and I used this to differentiate if the user has clicked on a root library key or a library in the library tree.